### PR TITLE
Change sonarqube scanner workspace name + tests

### PR DESCRIPTION
**Pull request by rvankruistum**
# Changes

Changes the source workspace name to `source` instead of `source-dir`. When running code-coverage analysis in pytest (possibly other tasks) the workspace dir is hardcoded into the coverage file. This change makes sure the source workspace path sonarqube-scanner is the same path as in the coverage file.

I also added a test (as there was none). This uses a public sonarcloud project as it's very resource intensive to run an entire sonarqube as a sidecar.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

### CODE REVIEW

File: task/sonarqube-scanner/0.2/sonarqube-scanner.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: sonarqube-scanner
+  labels:
+    app.kubernetes.io/version: "0.1"
```
Comment (vinamra28) [inline task/sonarqube-scanner/0.2/sonarqube-scanner.yaml:6]:
```suggestion
    app.kubernetes.io/version: "0.2"
```

File: task/sonarqube-scanner/0.2/tests/run.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: sonarqube-run
+spec:
+  pipelineRef:
+    name: sonarqube-pipeline
+  workspaces:
+    - name: shared-workspace
+      persistentvolumeclaim:
```
Comment (vinamra28) [inline task/sonarqube-scanner/0.2/tests/run.yaml:64]: nit
```suggestion
      persistentVolumeClaim:
```

**Review by vinamra28 — COMMENTED** [review]
/ok-to-test

**Comment by rvankruistum** [comment]
Random unrelated tests are failing... Just keep retrying I guess

**Comment by rvankruistum** [comment]
@vinamra28 Integration tests keep failing on unrelated tests. Any idea what's going on?

**Comment by vinamra28** [comment]
@rvankruistum I have raised a PR #922 which will unblock the CI

File: task/sonarqube-scanner/0.2/README.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
+# SonarQube
+
+SonarQube™ is the leading tool for continuously inspecting the Code Quality and Security™ of your codebases, all while empowering development teams. Analyze over 25 popular programming languages including C#, VB.Net, JavaScript, TypeScript and C++. It detects bugs, vulnerabilities and code smells across project branches and pull requests.
+
+The following task can be used to perform static analysis on the source code provided the SonarQube server is hosted.
+
+For creating your own `sonar-project.properties` please follow the guide [here](https://docs.sonarqube.org/latest/analysis/analysis-parameters/). Sample properties file can be found [here](./samples/sonar-project.properties)
```
Comment (vinamra28) [inline task/sonarqube-scanner/0.2/README.md:7]: @rvankruistum, sample `sonar-project.properties` doesn't exist anymore. Can you please update here as well?

File: task/sonarqube-scanner/0.2/README.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/sonarqube-scanner/0.1/sonarqube-scanner.yaml
```
Comment (vinamra28) [inline task/sonarqube-scanner/0.2/README.md:12]:
```suggestion
kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/sonarqube-scanner/0.2/sonarqube-scanner.yaml
```

File: task/sonarqube-scanner/0.2/README.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+```
+
+## Pre-requisite
+
+Install the `git-clone` task from the catalog
+
+```
+https://raw.githubusercontent.com/tektoncd/catalog/v1beta1/git/git-clone.yaml
```
Comment (vinamra28) [inline task/sonarqube-scanner/0.2/README.md:20]: maybe you can update this to point to latest one from main branch. WDYT?

**Comment by rvankruistum** [comment]
@vinamra28 Thanks for the comments. Everything should be good to go now 😺

**Comment by vinamra28** [comment]
@rvankruistum sorry to bother you again, can please squash the commits and add a better commit message?

**Comment by rvankruistum** [comment]
> @rvankruistum sorry to bother you again, can please squash the commits and add a better commit message?

Done

**Review by vinamra28 — APPROVED** [review]

**Comment by rvankruistum** [comment]
Another unrelated test failed. This is getting a bit annoying now.

**Comment by vinamra28** [comment]
/retest
sorry for the inconvenience. Hope it will get fixed now
